File: js/connect_april.js
const fs = require('fs');

class ConnectApril {
  static recv_request() {
    const n = process.argv[2];
    const filePath = `.${n}.input`;

    try {
      const data = fs.readFileSync(filePath, 'utf8');
      return JSON.parse(data);
    } catch (err) {
      console.error(err);
      return null;
    }
  }

  static send_response(response) {
    const n = process.argv[2];
    const filePath = `.${n}.output`;

    try {
      fs.writeFileSync(filePath, JSON.stringify(response));
    } catch (err) {
      console.error(err);
    }
  }
}

File: dart/connect_april.dart
// april.dart

import 'dart:convert';
import 'dart:io';

class ConnectApril {
  static Map<String, dynamic>? recvRequest() {
    final n = int.parse(Platform.args.last!);
    final filePath = '.$n.input';

    try {
      final data = File(filePath).readAsStringSync();
      return json.decode(data);
    } catch (e) {
      print('Error reading file: $filePath');
      return null;
    }
  }

  static void sendResponse(Map<String, dynamic> response) {
    final n = int.parse(Platform.args.last!);
    final filePath = '.$n.output';

    try {
      File(filePath).writeAsStringSync(json.encode(response));
    } catch (e) {
      print('Error writing file: $filePath');
    }
  }
}

